extensions/native/compiler/src/ir/collections.rs

```diff
@@ -490,3 +557,35 @@ pub fn unsafe_array_transmute<C: Config, S, T>(arr: Array<C, S>) -> Array<C, T>
         unreachable!()
     }
 }
+
+#[allow(clippy::len_without_is_empty)]
+pub trait ArrayLike<C: Config> {
+    fn len(&self) -> Usize<C::N>;
+
+    fn ptr(&self) -> Ptr<C::N>;
+
+    fn is_fixed(&self) -> bool;
+
+    fn size_of(&self) -> usize;
```

Comment:
This is a bit confusing; at the call-site I interpreted this as size of the array type itself. Maybe rename to `element_size_of` or `size_of_element`?